distribution/lib/Standard/Table/0.0.0-dev/src/Data/Aggregate_Column.enso

```diff
@@ -177,29 +176,17 @@ type Aggregate_Column
         if this.new_name.is_nothing.not then this.new_name else
             get_name c = (this.resolve_column table c).name
             case this of
-                Group_By c _ -> (get_name c)
+                Group_By c _ -> get_name c
                 Count _ -> "Count"
-                Count_Distinct c _ _ ->
-                    case c of
-                        Vector.Vector _ -> "Count Distinct " + ((c.map get_name).join " ")
-                        _ -> "Count Distinct " + (get_name c)
-                Count_Not_Nothing c _ -> "Count Not Nothing " + (get_name c)
-                Count_Nothing c _ -> "Count Nothing " + (get_name c)
-                Count_Not_Empty c _ -> "Count Not Empty " + (get_name c)
-                Count_Empty c _ -> "Count Empty " + (get_name c)
-                Sum c _ -> "Sum " + (get_name c)
-                Average c _ -> "Average " + (get_name c)
-                Median c _ -> "Median " + (get_name c)
-                Percentile p c _ -> (p*100).floor.to_text + "%-ile " + (get_name c)
-                Mode c _ -> "Mode " + (get_name c)
-                Standard_Deviation c _ _ -> "Standard Deviation " + (get_name c)
-                Concatenate c _ _ _ _ _ -> "Concatenate " + (get_name c)
-                First c _ _ _ -> "First " + (get_name c)
-                Last c _ _ _ -> "Last " + (get_name c)
-                Maximum c _ -> "Maximum " + (get_name c)
-                Minimum c _ -> "Minimum " + (get_name c)
-                Shortest c _ -> "Shortest " + (get_name c)
-                Longest c _ -> "Longest " + (get_name c)
+                Count_Distinct columns _ _ ->
+                    as_vector = case columns of
+                        Vector.Vector _ -> columns
+                        _ -> [columns]
+                    "Count Distinct " + (as_vector.map get_name . join " ")
+                Percentile p c _ -> ((p*100).floor.to_text + "%-ile ") + get_name c
+                _ ->
+                    prefix = Meta.get_simple_type_name this . replace "_" " "
+                    prefix + " " + get_name this.column
```

Comment:
Wow :D I've got so mixed feelings about it - on one hand it's a super neat trick, on the other hand feels a bit hacky and like the former version was a bit clearer in what it was doing. But I guess this one is more future proof, so that's cool.